src/PrestaShopBundle/Controller/Admin/Sell/Order/InvoicesController.php

```diff
@@ -0,0 +1,149 @@
+     *
+     * @return array Template parameters
+     */
+    public function indexAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+
+        $byDateForm = $this->get('prestashop.admin.order.invoices.by_date.form_handler')->getForm();
+        $byStatusForm = $this->get('prestashop.admin.order.invoices.by_status.form_handler')->getForm();
+        $optionsForm = $this->get('prestashop.admin.order.invoices.options.form_handler')->getForm();
+
+        return [
+            'layoutTitle' => $this->trans('Invoices', 'Admin.Navigation.Menu'),
+            'requireAddonsSearch' => true,
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($legacyController),
+            'byDateForm' => $byDateForm->createView(),
+            'byStatusForm' => $byStatusForm->createView(),
+            'optionsForm' => $optionsForm->createView(),
+        ];
+    }
+
+    /**
+     * Action that generates invoices PDF by date interval
+     *
+     * @param Request $request
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller')~'_')", message="Access denied.")
+     *
+     * @return RedirectResponse
+     */
+    public function generatePdfByDateAction(Request $request)
+    {
+        $formHandler = $this->get('prestashop.admin.order.invoices.by_date.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
+                $this->flashErrors($errors);
+            }
+        }
+
+        return $this->redirectToRoute('admin_order_invoices');
+    }
+
+    /**
+     * Action that generates invoices PDF by order status
+     *
+     * @param Request $request
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller')~'_')", message="Access denied.")
+
+     * @return RedirectResponse
+     */
+    public function generatePdfByStatusAction(Request $request)
+    {
+        $formHandler = $this->get('prestashop.admin.order.invoices.by_status.form_handler');
+
+        $form = $formHandler->getForm();
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $data = $form->getData();
+
+            if ($errors = $formHandler->save($data)) {
+                $this->flashErrors($errors);
+            }
+        }
+
+        return $this->redirectToRoute('admin_order_invoices');
+    }
+
+    /**
+     * Process the Invoice Options configuration form.
+     *
+     * @param Request $request
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller')~'_')", message="Access denied.")
+     *
+     * @return RedirectResponse
+     */
+    public function processAction(Request $request)
+    {
+        $formHandler = $this->get('prestashop.admin.order.invoices.options.form_handler');
```

Comment:
Look like this function is called three times, only the form handler, the message and the redirect changed.